tests/unit/test_reservation.py

```diff
@@ -649,6 +657,256 @@ def test___no_reservation_order___get_connections___connections_returned_in_defa
     ]
 
 
+def test___multiplexer_session_not_created___get_connection_with_multiplexer___type_error_raised(
+    session_management_client: Mock,
+) -> None:
+    with ExitStack() as stack:
+        grpc_session_infos = create_nifake_session_infos(1)
+        grpc_multiplexer_session_infos = create_nimultiplexer_session_infos(1)
+        grpc_session_infos[0].channel_mappings.add(
+            pin_or_relay_name="Pin1",
+            site=2,
+            channel="3",
+            multiplexer_resource_name="Mux0",
+            multiplexer_route="route1",
+        )
+        reservation = MultiSessionReservation(
+            session_management_client, grpc_session_infos, grpc_multiplexer_session_infos
+        )
+        _ = stack.enter_context(reservation.initialize_session(_construct_session, "nifake"))
+
+        with pytest.raises(TypeError) as exc_info:
+            _ = reservation.get_connection_with_multiplexer(object, fake_driver.Session)
+
+        assert "Incorrect type for multiplexer session 'MyMultiplexer0'." in exc_info.value.args[0]
+        assert (
+            "Expected <class 'tests.utilities.fake_driver.Session'>, got <class 'NoneType'>."
+            in exc_info.value.args[0]
+        )
+
+
+def test___multiplexer_session_not_created___get_connection_with_multiplexer_with_multiplexer_session_type_object___connection_returned(
+    session_management_client: Mock,
+) -> None:
+    with ExitStack() as stack:
+        grpc_session_infos = create_nifake_session_infos(1)
+        grpc_multiplexer_session_infos = create_nimultiplexer_session_infos(1)
+        grpc_session_infos[0].channel_mappings.add(
+            pin_or_relay_name="Pin1",
+            site=2,
+            channel="3",
+            multiplexer_resource_name="Mux0",
+            multiplexer_route="route1",
+        )
+        reservation = MultiSessionReservation(
+            session_management_client, grpc_session_infos, grpc_multiplexer_session_infos
+        )
+        _ = stack.enter_context(reservation.initialize_session(_construct_session, "nifake"))
+
+        connection = reservation.get_connection_with_multiplexer(object, object)
+
+        assert connection.pin_or_relay_name == "Pin1"
+        assert connection.site == 2
+        assert connection.channel_name == "3"
+        assert connection.multiplexer_resource_name == "Mux0"
+        assert connection.multiplexer_route == "route1"
+        assert connection.multiplexer_session is None
+
+
+def test___no_connection___get_connection_with_multiplexer_with_multiplexer_session_type_object___value_error_raised(
```

Comment:
The test name is so confusing. Can we update?

Reply:
How about `test___no_connection___get_connection_with_multiplexer__raises_value_error`?

Reply:
I could see a similar test above with different criteria

Reply:
Yes, that has a different precondition which is there are connections but no multiplexer session isn't created.
In this case, there's no connection at all present for querying.

Reply:
Sure then we can proceed with your suggestion